src/main/java/org/candlepin/resource/EnvironmentResource.java

```diff
@@ -305,10 +305,15 @@ public ConsumerDTO createConsumerInEnvironment(String envId, ConsumerDTO consume
         this.validator.validateCollectionElementsNotNull(consumer::getInstalledProducts,
             consumer::getGuestIds, consumer::getCapabilities);
 
-        Environment e = lookupEnvironment(envId);
-        consumer.setEnvironment(translator.translate(e, EnvironmentDTO.class));
+        List<EnvironmentDTO> toUpdateEnvSet = new ArrayList<>();
```

Comment:
All the changes in this method are unneeded, because this is the legacy API endpoint (subman/katello will start using the main POST /consumers endpoint when registering with environments). There will be a different set of changes needed as part of another task for backwards compatibility (supporting old submans that  will be sending the list of environments as a CSV list in the envId string).

Reply:
Reverted these changes and added a TODO for the next task.